text/0000-dot-profile-utility-buildpack.md

```diff
@@ -0,0 +1,129 @@
+# Meta
+[meta]: #meta
+- Name: `.profile` Utility Buildpack
+- Start Date: 2022-01-25
+- Author(s): mboldt
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+As part of RFC 93, [`.profile` scripts will cease to be supported by the platform API](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#appprofile).
+This RFC proposes developing a [utility buildpack](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) to support `.profile` scripts to prevent regressions after RFC 93 is implemented.
+
+# Definitions
+[definitions]: #definitions
+
+*utility buildpack*: A buildpack officially supported by the Buildpack Authors' Tooling Team per [RFC 97](https://github.com/buildpacks/rfcs/blob/main/text/0097-official-utility-buildpacks.md) that provides a generic capability.
+
+# Motivation
+[motivation]: #motivation
+
+[RFC 93](https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md) resolves to remove shell-specific logic from the CNB Specification.
+Part of this includes removing support for `.profile` script in a future version of the Platform API.
+RFC 93 recommends supporting the `.profile` script functionality in a utility buildpack to avoid regressions.
+This proposal is to develop and support the `.profile` utility buildpack, allowing RFC 93 to be implemented without regression.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The target persona is a platform operator or implementor who wants to update to the latest platform API, while maintaining the `.profile` functionality for application developers.
+
+We propose developing and supporting a buildpack to provide an identical interface to the existing `.profile` functionality.
+It will:
```

Comment:
Can it also support a `.profile.d/` dir with arbitrary scripts?

I'm not sure we _need_ this (because in the past it's recommend for use by buildpacks only, which can use exec.d)

https://devcenter.heroku.com/articles/buildpack-api#profile-d-scripts

Reply:
Certainly seems technically feasible, but per the spec and the recommendation in the link you shared, I think it makes sense to stick with the principle of "`.profile.d/` is for buildpacks and `.profile` is for apps." I would expect a spec change to explicitly support `.profile.d/` scripts in apps, and that feels beyond the scope of this RFC.

Reply:
Thinking through a little more, maybe your point is less about apps using `.profile.d/` and more about supporting buildpacks that haven't moved from `.profile.d/` to `exec.d/`. I.e., could this unlock buildpacks that use `.profile.d/` to still work when the shell is removed from the launcher?

Again, certainly seems technically feasible.

I think it's a matter of (1) what is the state of buildpacks using `exec.d` vs `.profile.d` (if most are using `exec.d` it's less of an issue), and (2) do we want shell removal to be a forcing function for buildpack authors to migrate to `exec.d`? I don't have insight into these issues. What do you and others think?

Reply:
Let's move this RFC forward without `.profile.d` and revisit it later as an enhancement.